### DISCUSSION TITLE
Allow previous_tag specification on Release Form Automation

### DISCUSSION THREAD

**Discussion by thejeff77** [community/community#29048] [Other Feature Feedback, Questions, & Ideas]
Would like to request an additional parameter `previous_tag` 

Be added to the release form automation urls: https://docs.github.com/en/repositories/releasing-projects-on-github/automation-for-release-forms-with-query-parameters.

This would allow it to change from defaulting to `auto` to the specified tag in the URL.

Also consider allowing the passage of `generate_release_notes=1` in a queryparam to do this as part of the page load.

This would come in handy for automation since the default generation only goes back to the last semantic version tag, even though the last formal release in the release list may have been well before the previous tag. 

Eg: last release was at `v1.5.1`, when adding a release for existing tag `v1.8.1`, the `auto` option will diff back to `v1.8.0` instead of `v1.5.1`. This is arguably ambiguous, and I'm not saying the auto calculation is wrong. Would be nice to be able to specify the previous tag however.

Thanks for any consideration!

**Comment by ThePeterDenham** [comment]
This is a lovely addition that helps with our automation and prevents incorrect release notes from being generated. It also doesn't seem like a massive piece of work. It's a shame that nothing has changed since this issue was raised in 2022.

**Comment by varchasvakhare2022** [comment]
Hi @thejeff77,

Thanks for sharing this suggestion — you’ve described a very real limitation with the current Release Form automation.

At the moment, GitHub does not support a `previous_tag` query parameter, and the release form will always fall back to the automatically detected “latest semantic version” instead of the latest published release. This is expected behavior, but as you mentioned, it can produce incorrect diffs in workflows such as:

- Last release: `v1.5.1`
- Creating a release for existing tag: `v1.8.1`
- `auto` diff incorrectly jumps back to `v1.8.0` instead of `v1.5.1`

Being able to specify something like:

would give users explicit control and avoid ambiguity when the repo history doesn’t follow a strictly sequential tag pattern.

I've passed this along internally as a feature request so the team can review it. While I can’t promise a timeline, well-documented use cases like yours are extremely helpful for shaping future improvements.

Thanks again for taking the time to outline the need — it’s a solid enhancement idea and now on the team’s radar.
